### PR TITLE
cluster-api-provider-kubemark: add new cluster-api provider

### PR DESCRIPTION
**Pull request by paulfantom**
Add new repository to CI system.

/cc @enxebre @ingvagabund

### CODE REVIEW

File: ci-operator/config/openshift/cluster-api-provider-kubemark/openshift-cluster-api-provider-kubemark-master.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+promotion:
+  name: '4.0'
+  namespace: ocp
+base_images:
+  base:
+    name: '4.0'
+    namespace: ocp
+    tag: base
+binary_build_commands: NO_DOCKER=1 make build
+canonical_go_repository: github.com/openshift/cluster-api-provider-aws
```
Comment (ingvagabund) [inline ci-operator/config/openshift/cluster-api-provider-kubemark/openshift-cluster-api-provider-kubemark-master.yaml]: cluster-api-provider-kubemark

File: ci-operator/jobs/openshift/cluster-api-provider-kubemark/openshift-cluster-api-provider-kubemark-master-presubmits.yaml [hunk 2]
```diff
@@ -0,0 +1,230 @@
+presubmits:
+  openshift/cluster-api-provider-kubemark:
+  - agent: kubernetes
+    always_run: true
+    branches:
+    - master
+    context: ci/prow/e2e-aws-operator
```
Comment (ingvagabund) [inline ci-operator/jobs/openshift/cluster-api-provider-kubemark/openshift-cluster-api-provider-kubemark-master-presubmits.yaml]: e2e-kubemark-operator?

**Comment by enxebre** [comment]
we want to include the jenkins kubemark test here

**Comment by paulfantom** [comment]
@enxebre currently such job does not exist.

**Comment by enxebre** [comment]
cc @droslean

**Pull request merged**